test/integration/integration_test.go

```diff
@@ -505,6 +506,13 @@ func (i *integrationTest) exerciseIncrementalBuild(tag, imageName string, remove
 	if checkOnBuild {
 		i.fileExists(containerID, "/sti-fake/src/onbuild")
 	}
+
+	if took := time.Since(start); took > docker.DefaultDockerTimeout {
+		// https://github.com/openshift/source-to-image/issues/301 is a
+		// case where incremental builds would get stuck until the
+		// timeout.
+		t.Errorf("Test took too long (%v), some operation may have gotten stuck waiting for the DefaultDockerTimeout (%v). Inspect the logs to find operations that took long.", took, docker.DefaultDockerTimeout)
+	}
```

Comment:
> Worry: such an integration test would be meaningful, but notably slow.
The way to implement this would clearly be by decreasing the timeout used for the duration of the test run.

I think the only purpose of the timeout is to protect against potential bad behaviour from the docker daemon (although I would personally prefer us not to do this).  Either way, I think the existence of the timeout must not prevent end users from having builds or hooks or scripts which run for longer than the timeout.  This is one of the things I tried to improve before, which explained why previously in #301 it happened to hang for 2 minutes, and subsequently it hung forever.  We do not use the timeout in attach, pull or wait docker calls for this reason.

So I mean always succeed.